### PR TITLE
TRT5-friendly ONNX exporter 

### PR DESCRIPTION
**Pull request by drnikolaev**
This PR addresses  #21727 and replaces Gather-by-axis-0 cases with shorter Slice operators making output onnx files to be TRT5 and ONNX Runtime compatible. Also, minor issues with million annoying warnings have been fixed. As a bonus track, Filler Op now really asserts input tensor to be 1D.

### CODE REVIEW

**Comment by drnikolaev** [comment]
@houseroad

**Comment by drnikolaev** [comment]
Also, this would be terrific to put these tests back to CI, see #21727 for details.

**Comment by houseroad** [comment]
Sorry about the delay, will check this PR soon.

**Review by houseroad — CHANGES_REQUESTED** [review]
Thanks a lot for sending this PR.

It did multiple things together. Some changes are good to land, and some need more polish. I would like to split the changes into small PRs: 1) filler_op change, 2) ignore the warning (I would like to know what warnings are suppressed), 3) skip tests, 4) constant folding change, 5) gather -> slice change. 
This split will make reviewing and landing code easier.

File: caffe2/python/brew.py [hunk 1]
```diff
@@ -94,7 +96,9 @@ def scope_wrapper(*args, **kwargs):
                 "Or you can provide it in kwargs as model=<your_model>.")
                 new_kwargs = copy.deepcopy(model.arg_scope)
             func = self._registry[helper_name]
-            var_names, _, varkw, _= inspect.getargspec(func)
+            with warnings.catch_warnings():
```
[the remark this answers is not in the document]
Comment (drnikolaev) [reply caffe2/python/brew.py]: Deprecation warning. Lot's of them. I'll put a comment in a separate PR.

File: test/onnx/test_pytorch_onnx_caffe2.py [hunk 2]
```diff
@@ -1442,6 +1444,21 @@ def forward(self, input):
...
+    E                   tensor_type {
+    E                     elem_type: 1
+    E                     shape {
+    E                       dim {
+    E           -             dim_value: 0
+    E           ?                        ^
+    E           +             dim_value: 2
+    E           ?                        ^
+    """
+    @skipIfEmbed  # TODO - error above in embedded case only
```
[the remark this answers is not in the document]
Comment (drnikolaev) [reply test/onnx/test_pytorch_onnx_caffe2.py]: No,it does not pass in master, checked today:
```
$ ./scripts/onnx/test.sh
....
======================================= 22 failed, 1012 passed, 473 skipped, 7 deselected, 11855 warnings in 475.59 seconds ========================================
```
I could investigate why but let's first make the world green then cherry-pick and fix those 22 bad guys. Makes sense?

File: torch/csrc/jit/passes/onnx/constant_fold.cpp [hunk 3]
```diff
@@ -200,12 +200,175 @@ std::vector<Node*> getOnnxConstParentsToRemove(Node* node) {
...
 // constant-based computations/ops into an initializer node.
 void ConstantFoldONNX(Block* b, ParamMap& paramsDict) {
   AT_ASSERT(b->param_node());
+  /*
+   * We can do better for *static* cases like the one below.
+   * This use case is quite popular, but default implementation doesn't go deep.
+   * This handler replaces 5 ops shown below with just one:
+   *
+...
+%30 : Float(2, 256, 6, 6) = onnx::AveragePool[kernel_shape=[1, 1],strides=[1, 1]](%29), scope:
```
[the remark this answers is not in the document]
Comment (drnikolaev) [reply torch/csrc/jit/passes/onnx/constant_fold.cpp]: @houseroad @spandantiwari - this is actually the main thing breaking PT-exported ONNX. It feels like hacky but it's localized. I thought about to make it on demand feature but passing boolean switch to this place seems to be too much either. I'd love to hear your ideas about alternatives. These 5 operators are all over the place and they should be collapsed one way or another.

[the remark this answers is not in the document]
Comment (spandantiwari) [reply torch/csrc/jit/passes/onnx/constant_fold.cpp]: hi @drnikolaev - the current constant folding is written to fold only sub-tree starting with a leaf node. It seems like you are doing something similar in your CollectFoldables() method. If that's the intent then I agree with @houseroad that we should handle this case seamlessly in the existing code and fix any issues. If that is not the intent and this is not working with leaf nodes only, then I would suggest creating a new pass for this, and not try to handle everything here to keep the code design clean.

File: caffe2/python/rnn_cell.py [hunk 4]
```diff
@@ -179,7 +180,9 @@ def _rectify_apply_inputs(
             extra_inputs = _RectifyNames(extra_input_names)
             extra_inputs = zip(extra_input_names, extra_input_sizes)
 
-        arg_names = inspect.getargspec(self.apply_override).args
+        with warnings.catch_warnings():
```
[the remark this answers is not in the document]
Comment (drnikolaev) [reply caffe2/python/rnn_cell.py]: Fied in #23064

File: test/onnx/test_pytorch_onnx_caffe2.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (drnikolaev) [reply test/onnx/test_pytorch_onnx_caffe2.py]: #21727 for details

**Comment by garymm** [comment]
since there's been no response to @houseroad and @spandantiwari comments in over 2 years, I'm closing this.
@drnikolaev please feel free to open a new PR (ideally against onnx_ms_1) if you'd like to revive any of these changes.

**Pull request closed without merging**